### PR TITLE
[TACHYON-432] Make module for integration tests and move integration tests under this module from core and underfs

### PR DESCRIPTION
**Pull request by apc999**
https://tachyon.atlassian.net/browse/TACHYON-432

This PR moves most integration tests (except master/worker related, as they are not blocking TACHYON-423) in to a new module called integration-tests. The source file of integration tests are also renamed to be ClassNameIntegrationTest.java, in order to be different from ClassNameTest.java which is expected to be the unittests.

After this PR, integration-tests will by default use LocalFilesystemCluster. However, profile "hdfs" and "contractTest" are defined so that, 
`mvn test clean -PhdfsTest`, `mvn test clean -PglusterfsTest` or `mvn test clean -PcontractTest` will trigger integration tests with hdfs, glusterfs and hdfs-contract test.

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>hdfsTest</id>
```
Comment (calvinjia) [inline integration-tests/pom.xml]: Are we relying on `tachyon-client` to bring in `underfs-hdfs`?

File: integration-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    </profile>
+
+    <profile>
+      <id>contractTest</id>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-minicluster</artifactId>
+        </dependency>
+      </dependencies>
```
Comment (calvinjia) [inline integration-tests/pom.xml]: Shouldn't need this dependency anymore?

File: integration-tests/src/main/resources/META-INF/services/tachyon.underfs.UnderFileSystemFactory [hunk 3]
```diff
@@ -0,0 +1,2 @@
+tachyon.underfs.hdfs.HdfsUnderFileSystemFactory
+tachyon.underfs.glusterfs.GlusterFSUnderFileSystemFactory
```
Comment (calvinjia) [inline integration-tests/src/main/resources/META-INF/services/tachyon.underfs.UnderFileSystemFactory]: Why are these entries necessary? I thought these should be already be included in the underfs submodules?

File: integration-tests/src/test/java/tachyon/client/BlockInStreamIntegrationTest.java [hunk 4]
```diff
@@ -49,23 +58,24 @@ public static final void beforeClass() throws IOException {
     sLocalTachyonCluster = new LocalTachyonCluster(10000, 1000, Constants.GB);
     sLocalTachyonCluster.start();
     sTfs = sLocalTachyonCluster.getClient();
+    System.out.println(Constants.LOGGER_TYPE);
```
Comment (calvinjia) [inline integration-tests/src/test/java/tachyon/client/BlockInStreamIntegrationTest.java]: I think this line can be removed?

File: integration-tests/src/test/java/tachyon/client/BlockInStreamIntegrationTest.java [hunk 5]
```diff
@@ -23,13 +23,22 @@
...
 import tachyon.TestUtils;
+import tachyon.client.BlockInStream;
+import tachyon.client.EmptyBlockInStream;
+import tachyon.client.InStream;
+import tachyon.client.LocalBlockInStream;
+import tachyon.client.ReadType;
+import tachyon.client.RemoteBlockInStream;
+import tachyon.client.TachyonFS;
+import tachyon.client.TachyonFile;
+import tachyon.client.WriteType;
```
Comment (calvinjia) [inline integration-tests/src/test/java/tachyon/client/BlockInStreamIntegrationTest.java]: Are these imports necessary?

File: underfs/hdfs/pom.xml [hunk 6]
```diff
@@ -73,10 +73,6 @@
         <version>2.18.1</version>
         <configuration>
           <argLine>-Djava.security.krb5.realm= -Djava.security.krb5.kdc=</argLine>
```
Comment (calvinjia) [inline underfs/hdfs/pom.xml]: I think you can remove this entire plugin if the integration test are all in the new module.

**Comment by calvinjia** [comment]
@apc999 
Thanks for the patch! I did a pass over the changes, overall this is a massive improvement over the test structure we had in core before.

File: integration-tests/pom.xml [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>glusterfsTest</id>
+      <dependencies>
+        <dependency>
+          <groupId>org.tachyonproject</groupId>
+          <artifactId>tachyon-underfs-hdfs</artifactId>
```
Comment (calvinjia) [inline integration-tests/pom.xml]: Should this be `tachyon-underfs-glusterfs`?

File: integration-tests/pom.xml (hunk 1 above)
Comment (apc999) [inline integration-tests/pom.xml]: dependency on underfs-hdfs added

File: integration-tests/pom.xml (hunk 2 above)
Comment (apc999) [inline integration-tests/pom.xml]: Removed

File: integration-tests/src/test/java/tachyon/client/BlockInStreamIntegrationTest.java (hunk 4 above)
Comment (apc999) [inline integration-tests/src/test/java/tachyon/client/BlockInStreamIntegrationTest.java]: Good catch, removed

File: integration-tests/src/test/java/tachyon/client/BlockInStreamIntegrationTest.java (hunk 5 above)
Comment (apc999) [inline integration-tests/src/test/java/tachyon/client/BlockInStreamIntegrationTest.java]: Removed

File: integration-tests/pom.xml (hunk 7 above)
Comment (apc999) [inline integration-tests/pom.xml]: done

File: underfs/hdfs/pom.xml (hunk 6 above)
Comment (apc999) [inline underfs/hdfs/pom.xml]: Done, removed.

File: integration-tests/src/main/resources/META-INF/services/tachyon.underfs.UnderFileSystemFactory (hunk 3 above)
Comment (apc999) [inline integration-tests/src/main/resources/META-INF/services/tachyon.underfs.UnderFileSystemFactory]: Removed. It turns out by including tachyon-underfs-hdfs I don't really need this file.

Thanks for the catch.

File: integration-tests/pom.xml [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>contractTest</id>
```
Comment (haoyuan) [inline integration-tests/pom.xml]: why is this empty?

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**